frontend/packages/console-shared/src/hooks/perspective-utils.ts

```diff
@@ -0,0 +1,161 @@
+}
+
+type PerspectiveAccessReview = {
+  required?: AccessReviewResourceAttributes[];
+  missing?: AccessReviewResourceAttributes[];
+};
+
+type PerspectiveVisibility = {
+  state: PerspectiveVisibilityState;
+  accessReview?: PerspectiveAccessReview;
+};
+
+type Perspective = {
+  id: string;
+  visibility: PerspectiveVisibility;
+};
+
+export const getPerspectiveVisitedKey = (perspective: PerspectiveType): string =>
+  `${USERSETTINGS_PREFIX}.${PERSPECTIVE_VISITED_FEATURE_KEY}.${perspective}`;
+
+export const hasReviewAccess = async (accessReview: PerspectiveAccessReview) => {
+  let hasAccess: boolean = true;
+  const requiredAccessReview = accessReview?.required;
+  const missingAccessReview = accessReview?.missing;
+  const requiredPromises = requiredAccessReview?.map((resourceAttributes) =>
+    checkAccess(resourceAttributes),
+  );
+  const missingPromises = missingAccessReview?.map((resourceAttributes) =>
+    checkAccess(resourceAttributes),
+  );
+
+  requiredPromises?.length > 0 &&
+    (await Promise.all(requiredPromises)
+      .then((values) => {
+        // enable the prespective if all the access review checks are successful
+        hasAccess = hasAccess && values.every((val) => val?.status.allowed);
+      })
+      .catch((e) => {
+        // eslint-disable-next-line no-console
+        console.warn('AccessReview check failed', e);
+        hasAccess = false;
+      }));
+
+  missingPromises?.length > 0 &&
+    (await Promise.all(missingPromises)
+      .then((values) => {
+        // enable perspective if atleast one access review check fails
+        hasAccess = hasAccess && !values.every((val) => val?.status.allowed);
+      })
+      .catch((e) => {
+        // no update when a missing check fails
+        // eslint-disable-next-line no-console
+        console.warn('AccessReview check failed', e);
+      }));
+
+  return hasAccess;
+};
+
+export const usePerspectives = (): LoadedExtension<PerspectiveExtension>[] => {
+  const perspectiveExtensions = useExtensions<PerspectiveExtension>(isPerspective);
+  const [results, setResults] = React.useState<Record<string, boolean>>(() => {
+    let obj: Record<string, boolean> = {};
+    if (!window.SERVER_FLAGS.perspectives) {
+      obj = perspectiveExtensions.reduce(
+        (acc: Record<string, boolean>, ex: LoadedExtension<PerspectiveExtension>) => {
+          acc[ex.properties.id] = true;
+          return acc;
+        },
+        {},
+      );
+    } else {
+      const perspectives: Perspective[] = JSON.parse(window.SERVER_FLAGS.perspectives);
+      obj = perspectiveExtensions.reduce((acc, perspectiveExtension) => {
+        const perspective = perspectives?.find((p) => p.id === perspectiveExtension.properties.id);
+
+        if (perspective?.visibility?.state === PerspectiveVisibilityState.Disabled) {
+          acc[perspectiveExtension.properties.id] = false;
+        } else {
+          acc[perspectiveExtension.properties.id] = true;
+        }
+        return acc;
```

Comment:
We don't know if a perspective with access review can we read. I would only set perspectives to true that is not configured or Enabled. Wdyt?

```suggestion
        const perspective = perspectives?.find((p) => p.id === perspectiveExtension.properties.id);

        if (!perspective?.visibility?.state || perspective.visibility.state === PerspectiveVisibilityState.Enabled) {
          acc[perspectiveExtension.properties.id] = true;
        } else if (perspective?.visibility?.state === PerspectiveVisibilityState.Disabled) {
          acc[perspectiveExtension.properties.id] = false;
        }
        return acc;
```